pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -312,13 +389,14 @@ func (cm *certificateManager) getKeyPair(cli client.Client, secretName, secretNa
 	var issuer certificatemanagement.KeyPairInterface
 	if x509Cert.Issuer.CommonName == rmeta.TigeraOperatorCAIssuerPrefix {
 		if cm.keyPair.CertificateManagement != nil {
-			return certificateManagementKeyPair(cm, secretName, nil), nil, nil
+			return certificateManagementKeyPair(cm, secretName, secretNamespace, dnsNames), nil, nil
 		}
 		if string(x509Cert.AuthorityKeyId) == string(cm.AuthorityKeyId) {
 			issuer = cm.keyPair
 		} else {
 			if !readCertOnly {
 				// We want to return nothing, so a new secret will be created to overwrite this one.
+				cm.log.V(2).Info("KeyPair's authority key id doesn't match", "name", secretName)
```

Comment:
Similarly, perhaps make it warning level, and improve the comment to say that we make a new one.